src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -27,9 +30,9 @@
              comidi/routes->handler)
          settings
          path
-         wrap-with-authorization-check
-         puppet-version)))
-   context)
+         (fn [request] (wrap-with-authorization-check request {:oid-map oid-mappings}))
```

Comment:
I hear ya.  Adding a new service just for this feels like an overly big hammer.

I think I'm still leaning towards my initial suggestion of having the CA supply the OID-mapped auth middleware to the other services.  But maybe let's see what @nathanielksmith thinks before proceeding.